pyproject.toml

```diff
@@ -46,7 +46,7 @@ test = [
     "pytest-cov == 4.0.0",
 ]
 dev = [
-    "ruff == 0.0.202",
+    "ruff >= 0.0.291",
```

Comment:
VS Code was complaining about `ruff == 0.0.202` being too old for the Ruff extension, so I updated it here - if you have any specific reason to keep it pinned to such an old version, we can definitely do that. Current version is 1.9, FWIW.